### PR TITLE
Add conversion `core::num::NonZeroUx`

### PR DESCRIPTION
**Pull request by daxpedda**
Adds convenience functions to convert `core::num::NonZeroUx` to `NonZero<UInt<LIMBS>>`.

### CODE REVIEW

File: src/non_zero.rs [hunk 1]
```diff
@@ -64,6 +68,41 @@ where
     }
 }
 
+impl<const LIMBS: usize> NonZero<UInt<LIMBS>>
+where
+    UInt<LIMBS>: Integer,
+{
+    /// Create a [`NonZero<UInt>`] from a [`NonZeroU8`] (const-friendly)
+    // TODO(tarcieri): replace with `const impl From<NonZeroU8>` when stable
```
Comment (tarcieri) [inline src/non_zero.rs:76]: For now you can add non-`const` impls of these as well

**Review by tarcieri — APPROVED** [review]
These look good in isolation, although it'd be good to have the `From` impls too

File: src/non_zero.rs (hunk 1 above)
Comment (daxpedda) [reply src/non_zero.rs:76]: I left the TODOs because of `const`. Should I remove them?
Done otherwise.

Comment (tarcieri) [reply src/non_zero.rs:78]: TODOs are good. Eventually this can be unified as a `const impl`.

**Comment by tarcieri** [comment]
Thanks!

**Pull request merged**